qt/python/mantidqt/widgets/algorithmselector/widget.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import absolute_import, print_function
+
+import re
+
+from qtpy.QtCore import QModelIndex, Signal
+from qtpy.QtWidgets import (QWidget, QPushButton, QComboBox, QTreeWidget, QVBoxLayout,
+                            QHBoxLayout, QCompleter, QTreeWidgetItem)
+
+from mantidqt.utils.qt import block_signals
+from .presenter import IAlgorithmSelectorView, SelectedAlgorithm
+
+
+def get_name_and_version_from_item_label(item_label):
+    """
+    Extract algorithm name and version from a tree widget item label.
+    It must have the format:
+
+        AlgorithmName v.n
+
+    where n in v.n is a number
+    :param item_label: A label of a TreeWidgetItem
+    :return: SelectedAlgorithm tuple or None if the format is wrong.
+    """
+    match = re.match(r'(\w+) v\.(\d+)', item_label)
+    if match:
+        return SelectedAlgorithm(name=match.group(1), version=int(match.group(2)))
+    return None
+
+
+class AlgorithmSelectorWidget(IAlgorithmSelectorView, QWidget):
```

Comment:
Can we import this widget in the `algorithmselector.__init__` so that a user can do `from mantidqt.widgets.algorithmselector import AlgorithmSelectorWidget` ? I think the extra `.widget` is unnecessary detail for user.